src/monodroid/jni/monodroid-glue.cc

```diff
@@ -980,14 +981,14 @@ MonodroidRuntime::init_android_runtime (MonoDomain *domain, JNIEnv *env, jclass
 		lookup_bridge_info (domain, image, &osBridge.get_java_gc_bridge_type (i), &osBridge.get_java_gc_bridge_info (i));
 	}
 
-	// TODO: try looking up the method by its token
-	MonoClass *runtime = utils.monodroid_get_class_from_image (domain, image, "Android.Runtime", "JNIEnv");
-	MonoMethod *method = mono_class_get_method_from_name (runtime, "Initialize", 1);
+	// Android.Runtime.JNIEnv
+	MonoClass *runtime = utils.monodroid_get_class (domain, image, managed_token_ids.android_runtime_jnienv);
+	abort_unless (runtime != nullptr, "INTERNAL ERROR: Unable to find class Android.Runtime.JNIEnv in the Mono.Android assembly");
+	jnienv = runtime;
 
-	if (method == nullptr) {
-		log_fatal (LOG_DEFAULT, "INTERNAL ERROR: Unable to find Android.Runtime.JNIEnv.Initialize!");
-		exit (FATAL_EXIT_MISSING_INIT);
-	}
+	// Android.Runtime.JNIEnv.Initialize
+	MonoMethod *method = mono_get_method (image, managed_token_ids.android_runtime_jnienv_initialize, runtime);
```

Comment:
This method is highly repetitive: lookup `method` by token, assert `method` is not null, store "somewhere", continue.

Is there some way we could turn this into a *loop*?

```c
MonoMethod *jnienv_initialize;

struct MethodLookupInfo {
    MonoMethod **target;
    const char *description;
    uint32_t token_id;
} methods[] = {
    { &jnienv_initialize, "Android.Runtime.JNIEnv.Initialize", managed_token_ids.android_runtime_jnienv_initialize },
    { &registerType, "Android.Runtime.RegisterJniNatives", managed_token_ids.android_runtime_jnienv_registerjninatives },
    // …
};

for (int i = 0; i < sizeof(methods)/sizeof(methods[0]); ++i) {
    *methods[i].target = mono_get_method (image, methods[i].token_id, runtime);
    if (*methods [i].target != nullptr)
        continue;
    abort (methods[i].description);
}
```

Reply:
I took a different approach, especially that I realized that we mustn't use token IDs in desktop builds, see the updated code.